File: sdks/java/lib/src/main/java/models/shared/Collection.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * Collection
 * A collection of items, including files and folders.
 * 
 * Currently, the only collection available
 * is the `favorites` collection.
 * 
 * The contents of a collection can be explored in a
 * similar way to which the contents of a folder is
 * explored.
**/
public class Collection {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("collection_type")
    public CollectionCollectionTypeEnum collectionType;
    public Collection withCollectionType(CollectionCollectionTypeEnum collectionType) {
        this.collectionType = collectionType;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public Collection withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("name")
    public CollectionNameEnum name;
    public Collection withName(CollectionNameEnum name) {
        this.name = name;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public CollectionTypeEnum type;
    public Collection withType(CollectionTypeEnum type) {
        this.type = type;
        return this;
    }
}

File: sdks/go/pkg/models/shared/collection.go
package shared

type CollectionCollectionTypeEnum string

const (
	CollectionCollectionTypeEnumFavorites CollectionCollectionTypeEnum = "favorites"
)

type CollectionNameEnum string

const (
	CollectionNameEnumFavorites CollectionNameEnum = "Favorites"
)

type CollectionTypeEnum string

const (
	CollectionTypeEnumCollection CollectionTypeEnum = "collection"
)

// Collection
// A collection of items, including files and folders.
//
// Currently, the only collection available
// is the `favorites` collection.
//
// The contents of a collection can be explored in a
// similar way to which the contents of a folder is
// explored.
type Collection struct {
	CollectionType *CollectionCollectionTypeEnum `json:"collection_type,omitempty"`
	ID             *string                       `json:"id,omitempty"`
	Name           *CollectionNameEnum           `json:"name,omitempty"`
	Type           *CollectionTypeEnum           `json:"type,omitempty"`
}
